rrfs-test/CMakeLists.txt

```diff
@@ -37,22 +39,23 @@ if(FV3_DYCORE)
       endif()   
 
       get_all_keys(rrfs_fv3jedi_tests caselist)
+      set(casedir "${CMAKE_CURRENT_BINARY_DIR}/rundir-rrfs_fv3jedi_2022052619")
+      set(src_casedir "${rrfs-test_data_local}/rrfs-data_fv3jedi_2022052619")
+      if (NOT EXISTS "${casedir}")
+        file(MAKE_DIRECTORY ${casedir})
+      endif()
+      file(CREATE_LINK ${src_casedir}/DataFix ${casedir}/DataFix SYMBOLIC)
+      file(CREATE_LINK ${src_casedir}/Data_static ${casedir}/Data_static SYMBOLIC)
+      file(CREATE_LINK ${src_casedir}/INPUT ${casedir}/INPUT SYMBOLIC)
+      file(CREATE_LINK ${src_casedir}/Data ${casedir}/Data SYMBOLIC)
+      file(CREATE_LINK ${CMAKE_SOURCE_DIR}/rrfs-test/testoutput ${casedir}/testoutput SYMBOLIC)
       foreach(case ${caselist})
          get_from_dictionary(rrfs_fv3jedi_tests case exe)
-         set(casedir "${CMAKE_CURRENT_BINARY_DIR}/rundir-${case}")
-         set(src_casedir "${rrfs-test_data_local}/rrfs-data_fv3jedi_2022052619")
-         if (NOT EXISTS "${casedir}")
-           file(MAKE_DIRECTORY ${casedir})
-         endif()
-         file(CREATE_LINK ${src_casedir}/DataFix ${casedir}/DataFix SYMBOLIC)
-         file(CREATE_LINK ${src_casedir}/Data_static ${casedir}/Data_static SYMBOLIC)
-         file(CREATE_LINK ${src_casedir}/INPUT ${casedir}/INPUT SYMBOLIC)
-         file(CREATE_LINK ${src_casedir}/Data ${casedir}/Data SYMBOLIC)
          file(COPY ${src_yaml}/${case}.yaml DESTINATION ${casedir} )
          set(target_test ${case})
          ecbuild_add_test( TARGET ${target_test}
                         MPI      80
-                        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/rundir-${target_test}
+                        WORKING_DIRECTORY ${casedir}
```

Comment:
Yes - all ctests are now run in the same casedir (one casedir for fv3-jedi and one casedir for mpas-jedi). I found that mpas-jedi does the same thing for their ctests when I was trying to debug the failing tests in #158.

Reply:
@SamuelDegelia-NOAA  I would suggest to stick to the practice that each job to be run in its' own rundir.  Otherwise, it might mess the process and make it more difficult to manage and debug if any problems occur.  Some common data could use symbolic link.

Reply:
Especially, when the ctests could be run simultaneously,  multiple jobs running in the same dir seems not safe to me.

Reply:
This change was made because the ctest for the GETKF solver needs the output from the GETKF observer ctest. An alternative would be to add logic specific to the split GETKF ctests in `rrfs-test/CMakeLists.txt` but I prefer to make things as generic as possible. 

With the implementation in this PR, all the ctests work as expected. The output from each ctest is saved in `build/rrfs-test/Testing/Temporary/LastTest.log` which makes it easy to track any potential errors. 

> Especially, when the ctests could be run simultaneously, multiple jobs running in the same dir seems not safe to me.

I just ran all rrfs-test ctests in parallel using `ctest -j8` and they all passed. 

> For clarification. In the current case dir, are there any files or just a directory structure? The latter is what is applied in jedi. They have a root rundir : .../test/ . And all IO are in directories in this "root" rundir without interacting to each other.

The current casedir links in the `data` input directory but there are also a few fix files copied into the casedir (e.g., `namelist.atmosphere`, `geovars.yaml`). The ctests read these files but they do not modify anything. The ctests output various files but I modified some of the yamls so that the output files are specific to those tests (e.g., `ana_getkf.2024-05-27_00.00.00.nc`).

Reply:
I don't mind setting it back to the original practice @TingLei-NOAA - it is an easy change. But just giving my two cents here.

Reply:
@SamuelDegelia-NOAA  I would suggest either to have different rundir for different runs/ctests or as JEDI does, have a "super" rundir without any files except for directories to keep a clean rundir. Otherwise, especially when different developers are working on this part, the problem might occur .

Reply:
Is the concern that each test could be outputting in the same directory simultaneously? Because they do output into the main casedir, but they are outputting different files and those should not overlap. Or is the concern about reading the input files at the same time? Because the previous method of linking in the same input `data/` directory for each test would have the same concern .

Reply:
I think as long as no files (including logs) are overwritten, it should be fine to use the same directory. As @SamuelDegelia-NOAA mentioned. the JCSDA mpasjedi tests have already used this strategy.

Reply:
Update: I tested running `ctest -j8` a few more times and all tests continued to pass except for one failure of `rrfs_mpasjedi_2024052700_getkf_solver`. This test failed because the observer test needs to run first.

Reply:
@SamuelDegelia-NOAA Looks good! 
I am always curious whether we can set dependencies when running ctests in parallel?

Reply:
@guoqing-noaa I believe this is an option. Here is an example that I found on [this page](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/testing/unit_testing.html) of the JEDI documentation: 

```
ecbuild_add_test( TARGET       test_fv3jedi_forecast_fv3-gfs_compare
                  TYPE         SCRIPT
                  COMMAND      ${CMAKE_BINARY_DIR}/bin/compare.py
                  ARGS         testoutput/forecast_fv3-gfs.run testoutput/forecast_fv3-gfs.ref 1.0e-3 0
                  TEST_DEPENDS test_fv3jedi_forecast_fv3-gfs )
```

I can implement this for the getkf_solver test so that all ctests can be run in parallel.

Reply:
Yes this functionality works. You can see here that it waits to start the solver ctest until after the observer test has passed: 

```
(eva) [Samuel.Degelia@hfe08 rrfs-test]$ ctest -j8
Test project /scratch1/BMC/zrtrr/Samuel.Degelia/RDASApp_dev/RDASApp/build/rrfs-test
    Start 1: rrfs_fv3jedi_hyb_2022052619
    Start 2: rrfs_fv3jedi_letkf_2022052619
    Start 3: rrfs_mpasjedi_2024052700_Ens3Dvar
    Start 4: rrfs_mpasjedi_2024052700_letkf
    Start 5: rrfs_mpasjedi_2024052700_getkf
    Start 6: rrfs_mpasjedi_2024052700_getkf_observer
    Start 8: rrfs_mpasjedi_2024052700_bumploc
1/8 Test #2: rrfs_fv3jedi_letkf_2022052619 .............   Passed   46.69 sec
2/8 Test #1: rrfs_fv3jedi_hyb_2022052619 ...............   Passed  109.14 sec
3/8 Test #3: rrfs_mpasjedi_2024052700_Ens3Dvar .........   Passed  221.35 sec
4/8 Test #4: rrfs_mpasjedi_2024052700_letkf ............   Passed  268.31 sec
5/8 Test #6: rrfs_mpasjedi_2024052700_getkf_observer ...   Passed  299.60 sec
    Start 7: rrfs_mpasjedi_2024052700_getkf_solver
6/8 Test #5: rrfs_mpasjedi_2024052700_getkf ............   Passed  312.33 sec
7/8 Test #7: rrfs_mpasjedi_2024052700_getkf_solver .....   Passed   55.73 sec
8/8 Test #8: rrfs_mpasjedi_2024052700_bumploc ..........   Passed  415.90 sec

100% tests passed, 0 tests failed out of 8

Label Time Summary:
mpi            = 1729.04 sec*proc (8 tests)
rdas-bundle    = 1729.04 sec*proc (8 tests)
script         = 1729.04 sec*proc (8 tests)

Total Test time (real) = 415.92 sec
```